src/RESTController.js

```diff
@@ -159,27 +159,28 @@ const RESTController = {
         headers[key] = customHeaders[key];
       }
 
-      function handleProgress(event) {
+      function handleProgress(type, event) {
         if (options && typeof options.progress === 'function') {
           if (event.lengthComputable) {
             options.progress(
+              type,
               event.loaded / event.total,
               event.loaded,
               event.total
```

Comment:
we could keep the 3 first parameters as is for backward compatibility, then we could add an object that can handle any future addition. we would just give the original event in it.
By testing original event.target people could infere if it's upload or download phase.
```
event.loaded / event.total,
event.loaded,
event.total,
{originalEvent:event}
```
But i'm ok with 
```
event.loaded / event.total,
event.loaded,
event.total,
type
```

and with
```
event.loaded / event.total,
event.loaded,
event.total,
event
```
for example.

Reply:
for determining the type of progress, we can use `event.target`, right?

Reply:
Yes, thats right.

Reply:
Upon checking, I tried logging `event` and here's what I got:
![image](https://user-images.githubusercontent.com/32805276/78658173-67fdcf00-78fc-11ea-98f4-fa40ede194f4.png)
I could check the instance, but it would be better to tell explicitly rather then leaving us the logic for checking whether it came from `download` or `upload`

Reply:
yes. I agree with that.
Maybe satisfying and scallable solution would be
`event.loaded / event.total,
event.loaded,
event.total,
{type:type}`
as arguments of the progress function.

Reply:
@ladigitale I see what you are trying to do now. I didn't realize there was a difference. I agree on the backwards compatibility by adding the options to the end.

There are other places in the code that you can add progress for instance file.getData in Parse.File etc